fastlane_core/spec/build_watcher_spec.rb

```diff
@@ -0,0 +1,154 @@
+        active?: false,
+        ready_to_submit?: true,
+        export_compliance_missing?: false,
+        train_version: '1.0',
+        build_version: '1',
+        upload_date: 1
+      )
+    end
+    let(:export_compliance_required_build) do
+      double(
+        'Export Compliance Required Build',
+        processed?: true,
+        active?: false,
+        ready_to_submit?: false,
+        export_compliance_missing?: true,
+        train_version: '1.0',
+        build_version: '1',
+        upload_date: 1
+      )
+    end
+
+    it 'returns an already-active build' do
+      expect(Spaceship::TestFlight::Build).to receive(:all_processing_builds).and_return([])
+      expect(Spaceship::TestFlight::Build).to receive(:latest).and_return(active_build)
+      expect(Spaceship::TestFlight::Build).to receive(:builds_for_train).and_return([active_build])
+
+      expect(UI).to receive(:success).with("Build #{active_build.train_version} - #{active_build.build_version} is already being tested")
+      found_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: 'some-app-id', platform: :ios)
+
+      expect(found_build).to eq(active_build)
+    end
+
+    it 'returns a ready to submit build' do
+      expect(Spaceship::TestFlight::Build).to receive(:all_processing_builds).and_return([])
+      expect(Spaceship::TestFlight::Build).to receive(:latest).and_return(ready_build)
+      expect(Spaceship::TestFlight::Build).to receive(:builds_for_train).and_return([ready_build])
+
+      expect(UI).to receive(:success).with("Successfully finished processing the build #{ready_build.train_version} - #{ready_build.build_version}")
+      found_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: 'some-app-id', platform: :ios)
+
+      expect(found_build).to eq(ready_build)
+    end
+
+    it 'returns a export-compliance-missing build' do
+      expect(Spaceship::TestFlight::Build).to receive(:all_processing_builds).and_return([])
+      expect(Spaceship::TestFlight::Build).to receive(:latest).and_return(export_compliance_required_build)
+      expect(Spaceship::TestFlight::Build).to receive(:builds_for_train).and_return([export_compliance_required_build])
+
+      expect(UI).to receive(:success).with("Successfully finished processing the build #{export_compliance_required_build.train_version} - #{export_compliance_required_build.build_version}")
+      found_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: 'some-app-id', platform: :ios)
+
+      expect(found_build).to eq(export_compliance_required_build)
+    end
+
+    it 'waits when a build is still processing' do
+      expect(Spaceship::TestFlight::Build).to receive(:all_processing_builds).and_return([processing_build])
+      expect(Spaceship::TestFlight::Build).to receive(:builds_for_train).and_return([processing_build], [ready_build])
+      expect(FastlaneCore::BuildWatcher).to receive(:sleep).with(10)
```

Comment:
suggestion: change this to `expect(FastlaneCore::BuildWatcher).to receive(:sleep)`

That way, if we ever change how long we want it to sleep, we don't have to go back and update all the tests where we expected a particular number